File: #3.Java Professional Modules/#3.Java WEB/Java MVC Frameworks - Spring/#01.Spring Essentials/src/main/java/itsgosho/domain/repositories/UserRepository.java
package itsgosho.domain.repositories;

import itsgosho.domain.entities.User;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User,String> {

    User findUserByUsername(String username);
    User findUserByEmail(String email);
}

File: #3.Java Professional Modules/#5.JS Web/#02.ReactJS/Workshop/back-end/src/server/repositories/user.repository.js
import UserModel from '../domain/models/user.model';
import TokenModel from "../domain/models/jwt.token.model";


const UserRepository = {

    async save(data) {
        return new TokenModel(data).save();
    },

    async deleteById(id) {

        return TokenModel.deleteOne({'_id': id}).exec();
    },

    async findById(id) {
        return TokenModel.findById(id).exec();
    },

    async findTotal() {
        return TokenModel.count({}).exec();
    },

    /*-----*/

    async findByUsername(username) {
        return UserModel.findOne({username}).exec();
    },

    async findByEmail(email) {
        return UserModel.findOne({email}).exec();
    },
};

export default UserRepository;